adr/ADR-40.md

```diff
@@ -125,6 +126,46 @@ all reconnect options are respected.
 For most clients, that means having a `reconnect` method on the
 Client/Connection handle.
 
+#### Custom server pool
+
+Client should have a way to specify custom server pool for reconnection
+attempts. When specified, client should use only those servers for reconnection
+attempts, ignoring any advertised servers from the Server.
+
+New pool should be used for all reconnection attempts until the client is
+restarted or a new pool is specified. New pool is subject to the same rules as
```

Comment:
Need to clarify here what happens if a custom pool it set at run time and gossip learn is on, do they merge like they would normally from the initial url list or not?